scripts/jsonrpcproxy.py

```diff
@@ -0,0 +1,151 @@
+gcc -O3 -I /usr/include/python3.5m -o rpcproxy rpcproxy.c \
+-Wl,-Bstatic -lpython3.5m -lz -lexpat -lutil -Wl,-Bdynamic -lpthread -ldl -lm
+
+"""
+
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from os import path
+from urllib.parse import urlparse
+import socket
+import sys
+
+if sys.platform == 'win32':
+    import win32file
+    import pywintypes
+
+
+VERSION = '0.1.0a1'
+BUFSIZE = 32
+DELIMITER = ord('\n')
+INFO = """JSON-RPC Proxy
+
+Version: {}
+Proxy: {}
+Backend: {}
+"""
+
+
+class NamedPipe(object):
+    """Windows named pipe simulating socket."""
+
+    def __init__(self, ipc_path):
+        try:
+            self.handle = win32file.CreateFile(
+                ipc_path, win32file.GENERIC_READ | win32file.GENERIC_WRITE,
+                0, None, win32file.OPEN_EXISTING, 0, None)
+        except pywintypes.error as err:
+            raise IOError(err)
+
+    def recv(self, max_length):
+        (err, data) = win32file.ReadFile(self.handle, max_length)
+        if err:
+            raise IOError(err)
+        return data
+
+    def sendall(self, data):
+        return win32file.WriteFile(self.handle, data)
+
+    def close(self):
+        self.handle.close()
+
+
+def get_ipc_socket(ipc_path, timeout=1):
+    if sys.platform == 'win32':
+        return NamedPipe(ipc_path)
+
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.connect(ipc_path)
+    sock.settimeout(timeout)
+    return sock
+
+
+class HTTPRequestHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        self.send_response(200)
+        self.send_header("Content-type", "text/plain")
+        self.end_headers()
+        backend_url = 'unix:' + self.server.backend_address
+        proxy_url = '{}:{}'.format(self.server.server_name,
+                                   self.server.server_port)
+        info = INFO.format(VERSION, backend_url, proxy_url)
+        self.wfile.write(info.encode('utf-8'))
+
+    def do_POST(self):
+        request_length = int(self.headers['Content-Length'])
+        request_content = self.rfile.read(request_length)
+        # self.log_message("Headers:  {}".format(self.headers))
+        self.log_message("Request:  {}".format(request_content))
+
+        response_content = self.server.process(request_content)
+        self.log_message("Response: {}".format(response_content))
+
+        self.send_response(200)
+        self.send_header("Content-type", "application/json")
+        self.send_header("Content-length", len(response_content))
+        self.end_headers()
+        self.wfile.write(response_content)
+
+
+class Proxy(HTTPServer):
+
+    def __init__(self, proxy_url, backend_url):
+
+        proxy_url = urlparse(proxy_url)
+        assert proxy_url.scheme == 'http'
+        proxy_address = proxy_url.hostname, proxy_url.port
+
+        backend_url = urlparse(backend_url)
+        assert backend_url.scheme == 'unix'
```

Comment:
this now fails for me on windows

Reply:
It is not ready yet. I'm moving to Windows to finish it.